### PR TITLE
feat: LT-1670 - Session timeout

### PR DESCRIPTION
**Pull request by a-bek**

### CODE REVIEW

File: conf/settings.py [hunk 1]
```diff
@@ -220,3 +221,7 @@
 CSP_CONNECT_SRC = env("CSP_CONNECT_SRC")
 CSP_INCLUDE_NONCE_IN = env("CSP_INCLUDE_NONCE_IN")
 CSP_REPORT_ONLY = env("CSP_REPORT_ONLY")
+
+
+# Session timeout
+SESSION_TIMEOUT_SECONDS = 60 * 60
```
Comment (Mark-Spiers-kainos) [inline conf/settings.py]: should it not be 30 minutes?

File: conf/middleware.py [hunk 2]
```diff
@@ -58,3 +60,25 @@ def __call__(self, request):
...
+
+    def __call__(self, request):
+        start = request.session.get(SESSION_TIMEOUT_KEY, time.time())
+
+        timeout = getattr(settings, "SESSION_TIMEOUT_SECONDS", 3600)
+
+        if time.time() - start > timeout:  # session expired
+            request.session.flush()
+            logout(request)
+            return redirect(env("AUTHBROKER_URL") + "/sso/accounts/logout")
```
Comment (BenAndrew-Kainos) [inline conf/middleware.py]: Please define this URL in constants

File: conf/middleware.py [hunk 3]
```diff
@@ -58,3 +60,25 @@ def __call__(self, request):
...
+
+
+class SessionTimeoutMiddleware:
+    def __init__(self, get_response=None):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        start = request.session.get(SESSION_TIMEOUT_KEY, time.time())
+
+        timeout = getattr(settings, "SESSION_TIMEOUT_SECONDS", 3600)
```
Comment (BenAndrew-Kainos) [inline conf/middleware.py]: Please define constants

File: conf/settings.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply conf/settings.py]: Double check this

**Pull request closed without merging**